set_up/to-do-app-playbook.yml

```diff
@@ -0,0 +1,80 @@
+- name: Install to-do app on new web servers
+  hosts: webservers
+  remote_user: ec2-user
+
+- hosts: all
+  vars_prompt:
+  - name: boardKey
+    prompt: enter boardKey value
+
+
+  tasks:
+  - name: Install the latest versions of git and python
+    become: yes
+    yum:
+      name: git
+      state: latest
+      name: python3
+      state: latest
+
+  - name: Create to-do-app folder
+    ansible.builtin.file:
+      path: /opt/to-do-app
+      state: directory
+      owner: ec2-user
+      mode: '777'
+    become: yes
+
+  - name: Create to-do-app/set-up folder
+    ansible.builtin.file:
+      path: /opt/to-do-app/set-up
+      state: directory
+      owner: ec2-user
+      mode: '777'
+    become: yes  
+
+  - name: install poetry, if not already there
+    ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+    args:
+      creates: /home/ec2-user/.local/bin/poetry
+
+  - name: Create to-do-app/app folder
+    ansible.builtin.file:
+      path: /opt/to-do-app/app
+      state: directory
+      owner: ec2-user
+      mode: '777'
+    become: yes
+
+  - name: Git checkout
+    become: yes
```

Comment:
No need to be the root user to use git or to access the /opt/to-do-app folder
So you shouldn't have `become: yes`.

Minimise how much you do with elevated privileges